File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        max_len = 0
        n = len(s)
        
        for c in 'ABCDEFGHIJKLMNOPQRSTUVWXYZ':
            i = 0
            count = 0  
            
            for j in range(n):
                if s[j] != c:
                    count += 1
                
                while count > k:
                    if s[i] != c:
                        count -= 1
                    i += 1
                
                max_len = max(max_len, j - i + 1)
        
        return max_len

File: 0424-longest-repeating-character-replacement/0424-longest-repeating-character-replacement.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        int maxi = 0;
        int n = s.size();
        for (char c = 'A'; c <= 'Z'; c++) {
            int i = 0, j = 0;
            int count = 0; 
            while (j < n) {
                if (s[j] != c) {
                    count++;
                }
                while (count > k) {
                    if (s[i] != c) {
                        count--;
                    }
                    i++;
                }
                maxi = max(maxi, j - i + 1);
                j++;
            }
        }
        return maxi;
    }
};